internal/sidecar/service/volumereplication.go

```diff
@@ -290,3 +257,28 @@ func (rs *ReplicationServer) GetVolumeReplicationInfo(
 		LastSyncBytes:    resp.GetLastSyncBytes(),
 	}, nil
 }
+
+// setReplicationSource sets the replication source for the given volumeID or groupID.
+func setReplicationSource(src *csiReplication.ReplicationSource, req *proto.ReplicationSource) error {
+	if src == nil {
+		src = &csiReplication.ReplicationSource{}
+	}
+	if req == nil {
+		return fmt.Errorf("replication source is required")
+	}
+	if req.GetVolume() == nil && req.GetVolumegroup() == nil {
+		return fmt.Errorf("either volume or volume group is required")
+	}
+	if req.GetVolume() != nil {
+		src.Type = &csiReplication.ReplicationSource_Volume{Volume: &csiReplication.ReplicationSource_VolumeSource{
+			VolumeId: req.GetVolume().GetVolumeId(),
+		}}
+	}
+	if req.GetVolumegroup() != nil {
+		src.Type = &csiReplication.ReplicationSource_Volumegroup{Volumegroup: &csiReplication.ReplicationSource_VolumeGroupSource{
+			VolumeGroupId: req.GetVolumegroup().GetVolumeGroupId(),
+		}}
+	}
```

Comment:
```suggestion
        switch {
	case req.GetVolume() != nil :
		src.Type = &csiReplication.ReplicationSource_Volume{
		  Volume: &csiReplication.ReplicationSource_VolumeSource{
			VolumeId: req.GetVolume().GetVolumeId(),
		    } 
		  }
	case req.GetVolumegroup() != nil :
		src.Type = &csiReplication.ReplicationSource_Volumegroup{
		Volumegroup: &csiReplication.ReplicationSource_VolumeGroupSource{
			VolumeGroupId: req.GetVolumegroup().GetVolumeGroupId(),
		  }
		}
	}
```

This ensure we only set one of them ?

Reply:
AFAIC this can never be possible because its oneOf either one can be set but addressed the changes as requested